### PR TITLE
Added functionality to try last-matching pyparsing structure first #2064

### PR DESCRIPTION
**Pull request by Onager**
Added functionality to try last-matching pyparsing structure first

### CODE REVIEW

File: plaso/parsers/text_parser.py [hunk 1]
```diff
@@ -634,8 +634,11 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
 
       key = None
 
+      structures = self.LINE_STRUCTURES
```
[the remark this answers is not in the document]
Comment (Onager) [reply plaso/parsers/text_parser.py]: Changed this around a bit more

File: plaso/parsers/text_parser.py [hunk 2]
```diff
@@ -654,6 +657,12 @@ def ParseFileObject(self, parser_mediator, file_object, **kwargs):
           break
 
       if tokens and start == 0:
+        # Move matching key, structure pair to the front of the list, so that
+        # structures that are more likely to match are tried first.
+        if index != 0:
```
[the remark this answers is not in the document]
Comment (Onager) [reply plaso/parsers/text_parser.py]: Done

File: plaso/parsers/mac_wifi.py [hunk 3]
```diff
@@ -121,31 +125,31 @@ def __init__(self):
     self._last_month = 0
     self._year_use = 0
 
-  def _GetAction(self, function, text):
+  def _GetAction(self, action, text):
     """Parse the well known actions for easy reading.
 
     Args:
-      function (str): The function or action called by the agent.
+      action (str): The function or action called by the agent.
```
Comment (joachimmetz) [inline plaso/parsers/mac_wifi.py]: consistency nit, do not start with a capital after `:` also maybe reword to:
```
action (str): name of the function or action called by the agent.
```
